### PR TITLE
[Merged by Bors] - beacon: consider zero proposals a failure

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation

part of #3903

## Changes

- if the protocol finishes with zero proposals, the beacon protocol failed
- when beacon starts epoch, iterate all atxs just once (instead of twice, one for total weight/active set and one for weak coin unit allowance)
- retrieves atx header in one db read in handlers.go instead of getting atx ID (1 read) and getting atx header (max 1 read)
- tune some logging level and shorten messages. specifically 
  - using hex to log beacon proposals
  - drop log.binary()

### CODE REVIEW

**Comment by countvonzero** [comment]
bors try

**Review by dshulyak — APPROVED** [review]

File: beacon/handlers.go [hunk 1]
```diff
@@ -45,73 +46,71 @@ func (pd *ProtocolDriver) HandleWeakCoinProposal(ctx context.Context, peer p2p.P
...
+func (pd *ProtocolDriver) classifyProposal(logger log.Log, m ProposalMessage, atx *types.ActivationTxHeader, receivedTime time.Time) (category, error) {
 	minerID := m.NodeID.ShortString()
-	atxHeader, err := pd.cdb.GetAtxHeader(atxID)
-	if err != nil {
-		logger.Error("[proposal] failed to get ATX header", atxID, log.Err(err))
-		return invalid, fmt.Errorf("[proposal] failed to get ATX header (miner ID %v, ATX ID %v): %w", minerID, atxID, err)
-	}
-
-	atxTimestamp, err := atxs.GetTimestamp(pd.cdb, atxID)
+	// TODO: include timestamp in ATX header
```
Comment (fasmat) [inline beacon/handlers.go:99]: Part of this or the next PR?

**Review by fasmat — APPROVED** [review]

File: beacon/handlers.go (hunk 1 above)
Comment (countvonzero) [reply beacon/handlers.go:99]: next PR

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**